src/test/java/com/coravy/hudson/plugins/github/GithubLinkAnnotatorTest.java

```diff
@@ -25,13 +44,44 @@ public final void testAnnotateStringMarkupText() {
                         + "issues/9876/find'>close #9876</a> link");
     }
 
+    @Test
+    public final void testChangeSetAnnotateStringMarkupText() {
```

Comment:
I'm completely for the new good practices in tests! :)

Does not think that truncating a commit message on the server side is good idea. 
The commit message length is just a recommended approach, so can be ignored. With current UI jenkins can't provide good UX to see full change log with truncated messages. But commit sha mostly is just a way to fast jump to exact commit in github interface and its ok to limit it with 7 symbols